src/grpc/grpc_server.cc

```diff
@@ -1236,6 +1236,21 @@ CommonHandler::RegisterTrace()
     int32_t count;
     uint32_t log_frequency;
     std::string filepath;
+
+    if (!request.model_name().empty()) {
```

Comment:
I wounder if it would make sense to put this check and `TRITONSERVER_ServerModelConfig` into TraceManager::UpdateTraceSetting ([here](https://github.com/triton-inference-server/server/blob/8a823b2381f12f97219404543ad55b1437a62958/src/tracer.cc#L92) ). This way we only need to change `trace.cc` (cc @rmccorm4 )

Reply:
I was originally trying to put the changes there, however, I think it would require exposing `TRITONSERVER_ServerModelConfig` to the trace manager which would violate the hierarchy. Also, by the time `TraceManager::UpdateTraceSetting` is called, the server has already spent time parsing the input request. My ultimate decision to place the changes in the {endpoint}.cc files based on the structure of our `HandleModelConfig` functions, which, in the original ticket, was able to correctly identify the changes to the model repository.

Reply:
> which would violate the hierarchy.

Could you please clarify? I believe we already expose `TRITONSERVER_` API to our tracer: https://github.com/triton-inference-server/server/blob/8a823b2381f12f97219404543ad55b1437a62958/src/tracer.h#L50

Reply:
I was referring to the function `TRITONSERVER_ServerModelConfig`, itself which requires a `TRITONSERVER_Server` pointer parameter that the endpoints possess but `TraceManager` does not. The endpoints do not currently expose this pointer as far as I am aware.

Reply:
Ah, gotcha. Thank you for clarification.